Readme update (#3676)
openebs/openebs: README.md

# Welcome to OpenEBS
<BR>
We are an ultra-modern storage target platform, a Hyper-Converged storage software system and a modern Storage Data Fabric natively & tighty integrated into a Kubernetes platform.<BR>
<BR>
OpenEBS provides... <BR>
<BR>

- Stateful persistent storage voluems for Kubernetes
- is a 100% Cloud-Native storage solution
- Delivers a Kubernetes cluster-wide storage fabric that provides containers with access to storage across the entire Kubernetes cluster.
- Enterprise Grade data management capabilities such as **snapshots, clones and replicated volumes** <BR>
<BR>

OpenEBS is completely free and open source software. There are also commercial options available for organizations seeking enterprise support and management capabilities. These are provided by third-party vendors. For more information, see https://openebs.io <BR>
<BR>

> In 2022, OpenEBS introduced the MayaStor storage Data-Engine which is based on the Ultra High-Performance SPDK NVMe Stack, UIO-Ring Technology & Linux NVMEe drivers. This delivers higher performance than was previously possible in any Kubernetes storage product.<BR>

Reviewer: ```suggestion
In 2022, OpenEBS introduced the MayaStor storage Data-Engine which is based on the Ultra High-Performance SPDK NVMe Stack, IO URING Technology & Linux NVMe drivers. This delivers higher performance than was previously possible in any Kubernetes storage product.<BR>
```

> In 2022, OpenEBS introduced the MayaStor storage Data-Engine which is based on the Ultra High-Performance SPDK NVMe Kernel Stack, UIO-Ring Technology & Linux NVMEe drivers. This delivers higher performance than was previously possible in any Kubernetes storage product.<BR>

Reviewer: ```suggestion
In 2022, OpenEBS introduced the MayaStor storage Data-Engine which is based on the Ultra High-Performance SPDK NVMe Stack, io_uring Technology & Linux NVMe drivers. This delivers higher performance than was previously possible in any Kubernetes storage product.<BR>
```

<BR>
The OpeneBS project consists of multiple Data-Engines projects that are grouped into 2 EDITIONs.

> - The older storage engines are refered to as **LEGACY Edition**

Reviewer: ```suggestion
- The older storage engines are referred to as **LEGACY Edition**
```

- The ultra modern Mayastor Data-Engine is classifed as **STANDARD Edition**, which also includes...
    - LVM LocalPV
    - ZFS LocalPV
    - Device LocalPV
    - RawFile LocalPV
    - LocalPV-HostPath
    - LocalPV-Device

<BR>
The project plans to migrate, sunset and archive all LEGACY Data-Engines by June 2024. <BR>
<BR>

> All **LEGACY** Data-Engines will me tagged as DEPRECATED and will be moved to ARCHIVE status by June 2024. These Data-Engines are:

Reviewer: ```suggestion
All **LEGACY** Data-Engines will be tagged as DEPRECATED and will be moved to ARCHIVE status by June 2024. These Data-Engines are:
```

**Our vision** is simple: let storage and storage services for persistent workloads be fully integrated into the environment so that each team and workload benefits from the granularity of control and Kubernetes native behaviour.

> ## Roadmap (as at Jan 2024)

Reviewer: ```suggestion
## Roadmap (as of Jan 2024)
```

OpenEBS is 100% open source software. 
The project source code is spread across multiple repos: 
<BR>
<BR>

> The Roamap is focused exclusively on the modern Data-Engine **Mayastor**. It does not define any net-new features or capabilities for any OpenEBS LEGACY projects or projects that are tagged & deifned as DEPRERCATED or ARCHIVED. Currently those proejcts are defined as the follows (see refernces above for the details on thje projects DEPRECATED and ARCHIVAL strategy)...

Reviewer: ```suggestion
The Roamap is focused exclusively on the modern Data-Engine **Mayastor**. It does not define any net-new features or capabilities for any OpenEBS LEGACY projects or projects that are tagged & defined as DEPRERCATED or ARCHIVED. Currently those proejcts are defined as the follows (see refernces above for the details on the projects DEPRECATED and ARCHIVAL strategy)...
```

- Jiva
- cStor
- NFS-Provisioner
<BR>

**MayaStor Roadmap : 2024 Phase-2**

> - Forward facing planned release date, Release version numbers and feature priotiries re subject to chage as the project Maintainers/Leadership/community continiously update and adjust the **Release Feature bundling Strategy** to react to K8s industry movements, trends and our community influence.

Reviewer: ```suggestion
- Forward facing planned release date, Release version numbers and feature priorities are subject to change as the project Maintainers/Leadership/community continuously update and adjust the **Release Feature bundling Strategy** to react to K8s industry movements, trends and our community influence.
```

|  ID  | Feature name                   | Description and user stpory                                            | Release, links, tracking issue, GitHub repo                                                   |
| :--- | :----------------------------- | :--------------------------------------------------------------------- | :------------------------------------------------------------------------------------------ |
| 1    | Multi-replica volume snapshot and CSI cloning | Able to take consistent snapshots across all available replicas of a volume                                     | Pri 1 /  Rel: (Q1 2024)    |

> | 2    | Volume resize                                 | Able to increase volume size and overlaying file-system size with I/O continuity                                | Pri 1 /  Rel: (Q1 2024)    |

Reviewer: ```suggestion
| 2    | Volume resize                                 | Able to increase volume size and overlaying filesystem size with I/O continuity                                | Pri 1 /  Rel: (Q1 2024)    |
```

| 3    | DiskPool resize                               | Able to increase pool capacity by expansion of underlying disk pool device(s) with I/O continuity               | Pri 1 /  Rel: (Q1 2024)    |

> | 4    | DiskPool media aggregation mgmt               | Able to create, expand & mannage virtual disks that are aggregated from multiple physical disks                 | Pri 1 /  Rel: (Q2 2024)    |

Reviewer: ```suggestion
| 4    | DiskPool media aggregation mgmt               | Able to create, expand & manage virtual disks that are aggregated from multiple physical disks                 | Pri 1 /  Rel: (Q2 2024)    |
```

| 6    | Local-PV Data-Engine integrated + enabled     | Dynamically provision a persistent volume of LocalPV (non-replicated) type using non-SPDK blobstor for storage  | Pri 1 /  Rel: (Q1 2024)    |
| 6    | Local-PV Data-Engine integrated + enabled     | Dynamically provision a persistent volume of LocalPV (non-replicated) type using non-SPDK blobstor for storage  | Pri 1 /  Rel: (Q1 2024)    |
| 6.1  | Local-PV Hostpath enabled                     | Able to provision a persistent volume of Local-PV (non-replicated) using type : K8s Hostpath addressed storage  | Pri 2 /  Rel: (Q2 2024)    |
| 6.2  | Local-PV Device enabled                       | Able to provision a persistent volume of Local-PV (non-replicated) using type : K8s Device addressed storage    | Pri 2 /  Rel: (Q2 2024)    |

> | 6.3  | Local-PV RawFile Soft Luns enabled            | Able to provision a persistent volume of Local-PV (non-replicated) using type : K8s Soft Fileystem lun addressed storage    | Pri 3 /  Rel: (Q3 2024)  |

Reviewer: ```suggestion
| 6.3  | Local-PV RawFile Soft Luns enabled            | Able to provision a persistent volume of Local-PV (non-replicated) using type : K8s Soft Filesystem lun addressed storage    | Pri 3 /  Rel: (Q3 2024)  |
```

| 6.4  | Local-PV RawFile Multi-Filesystem support     | Multi filesystems support for Local-PV RawFile Soft luns : ext3, ext4, XFS, BTRFS, f2fs, ZNS                    | Pri 3 /  Rel: (Q3 2024)   |

> | 6.5  | NDM integrated + enabled                      | NDM supoort for all Local-PV dependant services                                                                 | Pri 2 /  Rel: (Q2 2024)   |

Reviewer: ```suggestion
| 6.5  | NDM integrated + enabled                      | NDM support for all Local-PV dependant services                                                                 | Pri 2 /  Rel: (Q2 2024)   |
```

| 7    | HyperLocal-PV Data-Engine                     | Dynamically provision a non-replicated PV of Local-PV type via SPDK blobstor LVol as storage + NVMe target device |  Pri 2 /  Rel: (Q2 2024)   |

> | 7.1  | HyperLocal-PV : UBlock mode                   | Non-replicated PV of Local-PV type via UBlock kernel intgartion to SPDK blobstor LVol as storage                  |  Pri 2 /  Rel: (Q2 2024)   |

Reviewer: ```suggestion
| 7.1  | HyperLocal-PV : UBlock mode                   | Non-replicated PV of Local-PV type via UBlock kernel integration to SPDK blobstor LVol as storage                  |  Pri 2 /  Rel: (Q2 2024)   |
```

> | 7.2  | HyperLocal-PV : PCIe mode                     | Non-replicated PV of Local-PV type via PCIe-based NVMe kernel intgartion to SPDK blobstor LVol as storage         |  Pri 2.5 /  Rel: (Q2 2024)*   |

Reviewer: ```suggestion
| 7.2  | HyperLocal-PV : PCIe mode                     | Non-replicated PV of Local-PV type via PCIe-based NVMe kernel integration to SPDK blobstor LVol as storage         |  Pri 2.5 /  Rel: (Q2 2024)*   |
```

| 8    | GUI Mgmt Portal & Dashboard                   | Provision, manage, monitor Mayastor deployments with a RESTful GUI interface - @ parity with Shell & kubectl cmds | Pri 3 /  Rel: (Q3 2024)    |

> | 8.1  | GUI Mgmt Portal & Dashboard : On-Prem         | Mgmt portal & Dashbord deployed privately on-prem for air-gaped architetcures                                     | Pri 3 /  Rel: (Q3 2024)    |

Reviewer: ```suggestion
| 8.1  | GUI Mgmt Portal & Dashboard : On-Prem         | Mgmt portal & Dashboard deployed privately on-prem for air-gaped architectures                                     | Pri 3 /  Rel: (Q3 2024)    |
```

> | 8.1  | GUI Mgmt Portal & Dashboard : On-Prem         | Mgmt portal & Dashboard deployed privately on-prem for air-gaped architectures                                     | Pri 3 /  Rel: (Q3 2024)    |

Reviewer: ```suggestion
| 8.1  | GUI Mgmt Portal & Dashboard : On-Prem         | Mgmt portal & Dashboard deployed privately on-prem for air-gapped architectures                                     | Pri 3 /  Rel: (Q3 2024)    |
```

| 8.2  | GUI Mgmt Portal & Dashboard : In-Cloud SaaS   | Mgmt portal & Dashbord deployed as SaaS privately in-cloud for cloud enabled architetcures                        | Pri 3 /  Rel: (Q3 2024)    |

> | 8.3  | GUI Mgmt Portal & Dashboard : Global view     | Mgmt portal aggregated Global world  view of all k8s clusters configred to contribute anonymized global stats     | Pri 3 /  Rel: (Q3 2024)    |

Reviewer: ```suggestion
| 8.3  | GUI Mgmt Portal & Dashboard : Global view     | Mgmt portal aggregated Global world view of all k8s clusters configured to contribute anonymized global stats     | Pri 3 /  Rel: (Q3 2024)    |
```

| 9    | Storgae Encryption                            | Provision Encrypted data-at-rest volume via SPDK LVol layer - multi File system suppoort (ext3, ext4, XFS, BRFS)  | Pri 3 /  Rel: (Q3 2024)    |

> | 10   | Health & Supportability metrics + Dashboard   | Deep health diagnostics view of all elemets OpenEBS manages - enable Metric inclusion in Suppoort Bundle upoloads |  Pri 2.5 /  Rel: (Q2 2024*)   |

Reviewer: ```suggestion
| 10   | Health & Supportability metrics + Dashboard   | Deep health diagnostics view of all elements OpenEBS manages - enable Metric inclusion in Support Bundle upoloads |  Pri 2.5 /  Rel: (Q2 2024*)   |
```

> | 11   | E2E Storage UNMAP reclaim integation          | Support Discard: LINUX / UNMAP: SCSI / Deallocate: NVMe issued from file-system down to SPDK Blobstor elements    | Pri 3 /  Rel: (Q4 2024)    |

Reviewer: ```suggestion
| 11   | E2E Storage UNMAP reclaim integration          | Support Discard: LINUX / UNMAP: SCSI / Deallocate: NVMe issued from filesystem down to SPDK Blobstor elements    | Pri 3 /  Rel: (Q4 2024)    |
```

> | 12   | Thin provisioning phase-2                     | Thin Provision awarness and integarions with DiskPool metrics, pre-emptive intelligence actions                   | Pri 3 /  Rel: (Q4 2024)    |

Reviewer: ```suggestion
| 12   | Thin provisioning phase-2                     | Thin Provision awarness and integrations with DiskPool metrics, pre-emptive intelligence actions                   | Pri 3 /  Rel: (Q4 2024)    |
```

> | 12   | Thin provisioning phase-2                     | Thin Provision awarness and integrations with DiskPool metrics, pre-emptive intelligence actions                   | Pri 3 /  Rel: (Q4 2024)    |

Reviewer: ```suggestion
| 12   | Thin provisioning phase-2                     | Thin Provision awareness and integrations with DiskPool metrics, pre-emptive intelligence actions                   | Pri 3 /  Rel: (Q4 2024)    |
```

> | 13   | Native Object Store                           | An S3-compliant fast object store   inteatted with SPDK LVstore/LVols Blobstor & HyperLoca-PVl vols               | Pri 3 /  Rel: (Q4 2024)    |

Reviewer: ```suggestion
| 13   | Native Object Store                           | An S3-compliant fast object store   integrated with SPDK LVstore/LVols Blobstor & HyperLoca-PVl vols               | Pri 3 /  Rel: (Q4 2024)    |
```

> | 13   | Native Object Store                           | An S3-compliant fast object store   integrated with SPDK LVstore/LVols Blobstor & HyperLoca-PVl vols               | Pri 3 /  Rel: (Q4 2024)    |

Reviewer: ```suggestion
| 13   | Native Object Store                           | An S3-compliant fast object store   integrated with SPDK LVstore/LVols Blobstor & HyperLocal-PV vols               | Pri 3 /  Rel: (Q4 2024)    |
```

> | 14   | Zoned-SSD support                             | Integarted Western Digital Team's Mayastor ZNS feature for very high performance vols                             | Pri 2.5 /  Rel: (Q2 2024)   |

Reviewer: ```suggestion
| 14   | Zoned-SSD support                             | Integrated Western Digital Team's Mayastor ZNS feature for very high performance vols                             | Pri 2.5 /  Rel: (Q2 2024)   |
```
